src/rastervision/tagging/tasks/predict.py

```diff
@@ -101,40 +101,82 @@ def compute_probs(run_path, model, options, generator, split):
                 flip_img = np.fliplr(img)
                 aug_batch.append(np.expand_dims(flip_img, axis=0))
                 aug_batch = np.concatenate(aug_batch, axis=0)
-
-                aug_probs = np.mean(model.predict(aug_batch), axis=0,
-                                    keepdims=True)
-
-                y_probs.append(aug_probs)
-        # Otherwise, predicting probs on unaugmented images by batch
-        else:
+                aug_probs = model.predict(aug_batch)
+
+                for aug_ind in range(NUM_AUG):
+                    y_probs[aug_ind].append(aug_probs[aug_ind])
+
+            if (options.nb_eval_samples is not None and
+                    batch_ind * options.batch_size >= options.nb_eval_samples):
+                break
+
+        for aug_ind in range(NUM_AUG):
+            y_probs[aug_ind] = np.concatenate([y_probs[aug_ind]])
+            if options.nb_eval_samples is not None:
+                y_probs[aug_ind] = y_probs[aug_ind][0:options.nb_eval_samples]
+    # Otherwise, predicting probs on unaugmented images by batch
+    else:
+        y_probs = []
+        for batch_ind, batch in enumerate(split_gen):
             y_probs.append(model.predict(batch.x))
 
-        if (options.nb_eval_samples is not None and
-                batch_ind * options.batch_size >= options.nb_eval_samples):
-            break
+            if (options.nb_eval_samples is not None and
+                    batch_ind * options.batch_size >= options.nb_eval_samples):
+                break
 
-    y_probs = np.concatenate(y_probs, axis=0)
-    if options.nb_eval_samples is not None:
-        y_probs = y_probs[0:options.nb_eval_samples, :]
+        y_probs = np.concatenate(y_probs, axis=0)
 
-    np.save(probs_path, y_probs)
+        if options.nb_eval_samples is not None:
+            y_probs = y_probs[0:options.nb_eval_samples, :]
 
+    # Saving the probability numpy arrays
+    if split == TEST and options.test_augmentation:
+        for aug_ind in range(NUM_AUG):
+            probs_path = join(run_path, str(aug_ind) + '_' +
```

Comment:
Consider moving the code that generates the prob_path into a function since it is used in more than one place in this file.